src/rgw/rgw_zone.cc

```diff
@@ -790,7 +790,10 @@ void RGWZonePlacementInfo::decode_json(JSONObj *obj)
   JSONDecoder::decode_json("data_extra_pool", data_extra_pool, obj);
   uint32_t it;
   JSONDecoder::decode_json("index_type", it, obj);
-  JSONDecoder::decode_json("inline_data", inline_data, obj);
+  if(!JSONDecoder::decode_json("inline_data", inline_data, obj))
+  {
+    inline_data = true;
```

Comment:
thanks @umesh-mv, looks correct 👍

this isn't quite consistent with our coding style, though: https://github.com/ceph/ceph/blob/main/CodingStyle#L84-L103
```c++
  if (!JSONDecoder::decode_json("inline_data", inline_data, obj)) {
```

Reply:
thanks @cbodley,
corrected it.